spec/models/manageiq/providers/openstack/cloud_manager/event_catcher_spec.rb

```diff
@@ -1,7 +1,7 @@
 describe ManageIQ::Providers::Openstack::CloudManager::EventCatcher do
   before do
-    @ems = FactoryBot.create(:ems_openstack)
-    allow(@ems).to receive(:authentication_status_ok?).and_return(true)
+    _, _, zone = EvmSpecHelper.create_guid_miq_server_zone
+    @ems = FactoryBot.create(:ems_openstack, :with_authentication, :zone => zone)
     allow(ManageIQ::Providers::Openstack::CloudManager::EventCatcher).to receive(:all_ems_in_zone).and_return([@ems])
```

Comment:
NOTE I couldn't get rid of this one because we still need to stub `:event_monitor_available?` on the object but this is still exercising the new `all_valid_ems_in_zone?` logic because that is one level up from `all_ems_in_zone`